spec/factories/sponsors.rb

```diff
@@ -17,5 +17,42 @@
     organization { Organization.all.sample if sponsor_type.name == 'Organization' }
     payment_plan { Sponsor::PAYMENT_PLANS.sample }
     agent
+
+    trait :address_street do
+      address { "#{Faker::Address.building_number.to_s} #{Faker::Address.street_name}" }  
+    end
+    trait :email do
+      email { Faker::Internet.email }  
+    end
+    trait :contact1 do
+      contact1 { Faker::PhoneNumber.phone_number }
+    end
+    trait :contact2 do
+      contact2 { Faker::PhoneNumber.cell_phone }
+    end
+    trait :additional_info do
+      additional_info { Faker::Lorem.sentence }
+    end
+
+    #randomness: fields presence randomness is generated on each FactoryGirl.reload (or each rspec execution),
+    #            not on each FactoryGirl.build.
+    trait :random_optional_fields do  
+      [true,false].sample and address_street
+      [true,false].sample and email
+      [true,false].sample and contact1
+      [true,false].sample and contact2
+      [true,false].sample and additional_info
+    end
+
+    factory :sponsor_full, traits: [:address_street, :email, :contact1, :contact2, :additional_info]
+    factory :sponsor_random_optional_fields, traits: [:random_optional_fields] 
```

Comment:
I tried to also create a `:sponsor_random_optional_fields` factory that has the optional fields displayed or NOT(empty). Randomly.
Only problem with this is that the fields randomness only occurs once you reload `FactoryGirl` (on `$rspec` start or on a `FactoryGirl.reload`).

Randomness will not occur when you call 
`FactoryGirl.build(:sponsor_random_optional_fields)` and then you call again 
`FactoryGirl.build(:sponsor_random_optional_fields)`.
The same fields will be `nil` or `not nil`

I played with this for hours but no better solution.
I think it is still useful like this.
What do you think?